src/proxy/MultiMapProxy.ts

```diff
@@ -132,32 +132,37 @@ export class MultiMapProxy<K, V> extends BaseProxy implements MultiMap<K, V> {
         return this.encodeInvokeOnKey<number>(MultiMapValueCountCodec, keyData, keyData, 1);
     }
 
-    addEntryListener(listener: IMapListener<K, V>, key?: K, includeValue: boolean = true): Promise<string> {
+    addEntryListener(listener: EntryListener<K, V>, key?: K, includeValue: boolean = true): Promise<string> {
         const toObject = this.toObject.bind(this);
 
         /* tslint:disable: no-shadowed-variable */
-        const entryEventHandler = function (key: K, value: V, oldValue: V, mergingValue: V, event: number): void {
-            let parameters: any[] = [key, oldValue, value];
-            parameters = parameters.map(toObject);
-            let name: string;
+        const entryEventHandler = (key: K, value: V, oldValue: V, mergingValue: V, eventType: number, uuid: string,
+                                   numberOfAffectedEntries: number) => {
+            const member = this.client.getClusterService().getMember(uuid);
+            let entryEventParams: any[] = [this.name, key, value, oldValue, mergingValue, member];
```

Comment:
It exists here: https://github.com/hazelcast/hazelcast/blob/fd73a92b07d320007bdec04c25dcc073213e3b6a/hazelcast/src/main/java/com/hazelcast/core/AbstractIMapEvent.java#L26

Also, I decided to give to user entry event and map event objects rather than giving all fields.